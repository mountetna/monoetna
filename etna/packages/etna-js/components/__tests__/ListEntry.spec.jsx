import React from 'react';
import {render, screen, waitFor, fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import {ListEntryUpdatedColumn} from '../ListEntry';

describe('ListEntryUpdatedColumn', () => {
  it('renders correctly', async () => {
    const testObject = {
      updated_at: '2022-09-19T13:43:09+00:00',
      author: 'zeus@olympus.org|Zeus Almighty'
    };
    const testWidths = {};
    const {asFragment} = render(
      <ListEntryUpdatedColumn obj={testObject} widths={testWidths} />
    );

    await waitFor(() => screen.getByText(/by/));

    expect(screen.queryByText(/Zeus Almighty/)).toBeTruthy();
    expect(screen.queryByText(/zeus@olympus.org/)).toBeFalsy();
    expect(asFragment()).toMatchSnapshot();
  });
});
