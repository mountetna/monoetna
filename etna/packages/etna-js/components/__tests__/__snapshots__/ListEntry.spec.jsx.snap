// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListEntryUpdatedColumn renders correctly 1`] = `
<DocumentFragment>
  <div
    class="list-entry-column-group updated"
  >
    <div
      class="list-entry-updated-name"
    >
      Sep 19, 2022 by Zeus Almighty
    </div>
  </div>
</DocumentFragment>
`;
